### PR TITLE
nrf_cloud_agps: Add documentation

### PR DESCRIPTION
**Pull request by jtguggedal**
Adds documentation for nRF Cloud A-GPS library and sample application.

### CODE REVIEW

File: include/net/nrf_cloud_agps.h [hunk 1]
```diff
@@ -19,13 +19,31 @@
...
+ *
+ * @return 0 if successful, otherwise a (negative) error code.
+ */
 int nrf_cloud_agps_request_all(void);
 
-/**@brief Receive binary A-GPS data. */
+
+/**@brief Processes binary A-GPS data received from nRF Cloud.
+ *
+ * @param buf Poiner to data received from nRF Cloud.
```
Comment (jhn-nordic) [inline include/net/nrf_cloud_agps.h:40]: *pointer

File: include/net/nrf_cloud_agps.rst [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+###############
+
+Assisted GPS (A-GPS) is used to reduce the time for a GPS device to estimate its position, also called time to first fix.
+For a GPS to be able to get a position fix, it needs, for example, information about the satellite constellation, exact timing data and preferably a rough position estimate.
+The required information is broadcasted from the GPS satellites in a pattern that repeats every 12.5 minutes.
+With nRF Cloud A-GPS service, this broadcasted information can be downloaded from nRF Cloud, which is usually much faster.
+
+The nRF Cloud assisted GPS library provides a way to request and process A-GPS data from nRF Cloud to be used with nRF9160 SiP.
+
+To use the nRF Cloud A-GPS service, an nRF Cloud account is needed, and the device needs to be associated with a user's account.
```
Comment (jhn-nordic) [inline include/net/nrf_cloud_agps.rst:13]: Have this sentence in bold or something?

File: include/net/nrf_cloud_agps.rst [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+Practical considerations
+************************
+
+Which assistance data is needed at a certain time may vary, and requesting partial data set may then reduce data traffic, conserving both battery and data traffic.
+
+As an example, almanac data has a far longer validity (several months) than ephemeris data (2 to 4 hours).
+
+Receiving partial assistance data set may lead to the GPS downloading the missing data from satellites instead.
+
+When A-GPS data is downloaded using LTE network, the LTE link is in RRC conencted mode.
```
Comment (jhn-nordic) [inline include/net/nrf_cloud_agps.rst:41]: RRC *connected

File: include/net/nrf_cloud_agps.rst [hunk 4]
```diff
@@ -0,0 +1,60 @@
+.. _lib_nrf_cloud_agps:
+
+nRF Cloud A-GPS
+###############
+
+Assisted GPS (A-GPS) is used to reduce the time for a GPS device to estimate its position, also called time to first fix.
```
Comment (jhn-nordic) [inline include/net/nrf_cloud_agps.rst:6]: should write "(TTFF)" after "time to first fix" since "TTFF" is used later on

**Comment by jtguggedal** [comment]
@umapraseeda Thanks for reviewing! I've squashed the changes.

@jhn-nordic Could you revisit?

**Pull request merged**